index.js

```diff
@@ -1,21 +1,593 @@
+    }
+
+    let applicationKey =  core.getInput("application_key");
+    if (applicationKey) {
+      envVars[DD_APPLICATION_KEY] = applicationKey;
+    }
+
+
+    let platform = core.getInput("platform") || "ios";
+    platform = platform.toLowerCase();
+
+    const sdk = core.getInput("sdk") || getSDKForPlatform(platform);
+    const destination = core.getInput("destination") || getDestinationForPlatform(platform);
+    const configuration = core.getInput("configuration") || "Debug";
+    const libraryVersion = core.getInput("libraryVersion" || "");
+    const extraParameters = core.getInput("extraParameters") || "";
+
+
+    const itrEnabled = true;
+    if (itrEnabled) {
+      //If project uses testplan force use of code coverage
+      let file_list = recFindByExt(".", "xctestplan");
+      for (let testPlanFile of file_list) {
+        await deleteLinesContaining(testPlanFile, "codeCoverage");
+      }
+    }
+
+    //Create folder to store files
+    if (!fs.existsSync(sdkTestingDir)) {
+      fs.mkdirSync(sdkTestingDir);
+    }
+
+    //Read project
+    const workspace = await getWorkspace();
+    let xcodeproj = await getXCodeProj();
+    var projectParameter;
+
+    //download testing framework
+    await downloadLatestFramework(libraryVersion);
 
-    core.debug((new Date()).toTimeString()); // debug is only output if you set the secret `ACTIONS_RUNNER_DEBUG` to true
-    await wait(parseInt(ms));
-    core.info((new Date()).toTimeString());
+    if (workspace) {
+      console.log(`Workspace selected: ${workspace}`);
+      projectParameter = "-workspace " + `"${workspace}"`;
+    } else if (xcodeproj) {
+      console.log(`Project selected: ${xcodeproj}`);
+      projectParameter = "-project " + `"${xcodeproj}"`;
+    } else if (fs.existsSync("Package.swift")) {
+      console.log(`Package.swift selected`);
+      await swiftPackageRun(platform, extraParameters, itrEnabled);
+      return;
+    } else {
+      core.setFailed(
+        "Unable to find workspace, project or Swift package file. Please set with workspace or xcodeproj"
+      );
+    }
+
+    const scheme = await getScheme(workspace, xcodeproj);
+    console.log(`Scheme selected: ${scheme}`);
+
+      //copy configfile
+      const configfileName = "ddTesting.xcconfig";
+      const configFilePath = sdkTestingDir + "/" + configfileName;
+      createXCConfigFile(configFilePath, sdkTestingDir);
+
+
+
+    let codeCoverParam = "";
+    if (itrEnabled) {
+      codeCoverParam = "-enableCodeCoverage YES";
+    }
+
+    //build for testing
+    let buildCommand =
+      "xcodebuild build-for-testing " +
+      codeCoverParam +
+      " -xcconfig " +
+      configFilePath +
+      " " +
+      projectParameter +
+      " -configuration " +
+      configuration +
+      " -scheme " +
+      `"${scheme}"` +
+      " -sdk " +
+      sdk +
+      " -derivedDataPath " +
+      derivedDataPath +
+      " -destination " +
+      `"${destination}" ` +
+      extraParameters;
```

Comment:
something like this would be more readable I think: 
```suggestion
    const buildCommand = `xcodebuild build-for-testing ${codeCoverParam} ` +
    `-xcconfig ${configFilePath} ${projectParameter}` +
    `-configuration ${configuration} ` +
    `-scheme "${scheme}" `+
    `-sdk ${sdk} ` +
    `-derivedDataPath ${derivedDataPath} ` +
    `-destination "${destination}"` +
    extraParameters
```